feat: тесты по классу Burger

leanbre opened the pull request:
feat: тесты по Bun и Ingredient, потому что в задании было про 100% покрытие

leanbre committed:
feat: тесты по классу Burger
feat: тесты по Bun и Ingredient, потому что в задании было про 100% покрытие

okryukova commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,42 @@
+        this.price = price;
+    }
+
+    @Parameterized.Parameters
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,140 @@
+        // Создаем объект бургера
+        Burger newBurger = new Burger();
+        // Создаем ингредиенты соуса и котлеты для добавления
+        Ingredient ingredientSauce = new Ingredient(
```
⛔️Нужно исправить.  Нужно использовать моки

okryukova commented on src/test/java/IngredientTest.java:
```diff
@@ -0,0 +1,40 @@
+        // Проверяем, что тип ингредиента совпадает с заданным изначально
+        assertEquals(type, ingredient.getType());
+    }
+}
```
⛔️Нужно исправить.  Нужно приложить отчет о покрытии

leanbre committed:
fix: исправление теста moveIngredientTest на использование моков
feat: + отчет по покрытию тестами (согласно заданию, 100% ожидается на классе Burger)
